### PR TITLE
Fix queue size mismatch and add tests

### PR DESCRIPTION
**Pull request by conradgrobler**
The vsock implementation failed after 16 buffers were used in the receive queue. This happened because of a mismatch in the believed number of available descriptors.

When configuring the vhost vsock device, CrosVM uses the maximum queue size rather than the negotiated size (https://chromium.googlesource.com/chromiumos/platform/crosvm/+/d4505a7f1c9e4aa502ff49367863aedeadbafb9d/devices/src/virtio/vhost/worker.rs#74).

For now the vsock device just uses the CrosVM maximum queue size of 256. This is not compatible with QEMU, which has a maximum size of 128, but we use the virtio console comms channel on QEMU, rather than vsock.

This PR also adds some tests to ensure we go past the maximum queue size to detect similar issues in future.

### CODE REVIEW

File: experimental/virtio/src/vsock/mod.rs [hunk 1]
```diff
@@ -30,7 +30,11 @@ pub mod packet;
 pub mod socket;
 
 /// The number of buffer descriptors in each of the queues.
-const QUEUE_SIZE: usize = 16;
+///
+/// Note: We match the CrosVM max size for vhost vsock, as it seems CrosVM sets the number of
+/// descriptors to the maximum queue size rather than the negotiated size.
+/// See <https://chromium.googlesource.com/chromiumos/platform/crosvm/+/d4505a7f1c9e4aa502ff49367863aedeadbafb9d/devices/src/virtio/vhost/worker.rs#74>.
+const QUEUE_SIZE: usize = 256;
```
Comment (jul-sh) [inline experimental/virtio/src/vsock/mod.rs:37]: Out of curiosity, where does this number originate? The crosvm source says it's the max number the device supports. Is 256 from the spec, or from the linux implementation? :)

**Review by jul-sh — APPROVED** [review]
Thank you!

File: experimental/virtio/src/vsock/mod.rs (hunk 1 above)
Comment (conradgrobler) [reply experimental/virtio/src/vsock/mod.rs:37]: The device implementer (VMM) decides the maximum queue size. The only restrictions in the spec are that it must fit in a u16 and be a power of 2. CrosVM uses 256 and QEMU 128 for virtio vsock.

File: experimental/uefi/client/src/main.rs [hunk 2]
```diff
@@ -22,17 +22,21 @@ use std::io::{stdin, BufRead};
...
 #[derive(Parser, Debug)]
 struct Args {
-    /// address of the server
+    /// Address of the server.
     #[clap(long, default_value = "http://127.0.0.1:8000")]
     server: String,
 
-    /// message to be sent to the server. If not specified (with a response), uses stdio.
-    #[clap(long, requires = "expected-response")]
+    /// Message to be sent to the server. If not specified (with a response), uses stdio.
```
[the remark this answers is not in the document]
Comment (conradgrobler) [reply experimental/uefi/client/src/main.rs:29]: Thanks, I missed the review before I submitted. I will fix in a follow up PR.

File: experimental/uefi/client/src/main.rs [hunk 3]
```diff
@@ -22,17 +22,21 @@ use std::io::{stdin, BufRead};
...
+    /// Expected response from the server. If not specified (with a request), uses stdio.
+    #[clap(long, requires_all = &["request", "iterations"])]
     expected_response: Option<String>,
+
+    /// Number of times the request should be sent, and the expected response validated.
+    #[clap(long, requires_all = &["request", "expected-response"])]
+    iterations: Option<usize>,
 }
 
 async fn chat(client: &mut AttestationClient, message: String) -> anyhow::Result<String> {
```
[the remark this answers is not in the document]
Comment (conradgrobler) [reply experimental/uefi/client/src/main.rs:42]: It is very confusing. At the moment the naming is not well defined. At some point the bare-metal runtime will be the main Oak Functions runtime, but it is not yet very functional.

File: experimental/virtio/src/vsock/socket/tests.rs [hunk 4]
```diff
@@ -135,6 +135,34 @@ fn test_write_all() {
     assert_eq!(&data[MAX_PAYLOAD_SIZE..], second.get_payload());
 }
 
+#[test]
+fn test_many_echos() {
+    const DATA_LEN: usize = 47;
+    let (mut socket, transport) = new_socket_and_transport();
+    for i in 0..1000 {
+        let data = vec![17; DATA_LEN];
```
[the remark this answers is not in the document]
Comment (conradgrobler) [reply experimental/virtio/src/vsock/socket/tests.rs:143]: This is just value that is repeated in each item. Having a fixed value is not ideal though, so as part of the follow-up PR I will use some non-fixed values.
